pkg/securitycontext/util.go

```diff
@@ -81,42 +83,42 @@ func DetermineEffectiveSecurityContext(pod *v1.Pod, container *v1.Container) *v1
 		}
 	}
 
-	if containerSc.Capabilities != nil {
+	if runtime.GOOS != "windows" && effectiveSc != nil && containerSc != nil && containerSc.Capabilities != nil {
 		effectiveSc.Capabilities = new(v1.Capabilities)
 		*effectiveSc.Capabilities = *containerSc.Capabilities
 	}
 
-	if containerSc.Privileged != nil {
+	if runtime.GOOS != "windows" && effectiveSc != nil && containerSc != nil && containerSc.Privileged != nil {
 		effectiveSc.Privileged = new(bool)
 		*effectiveSc.Privileged = *containerSc.Privileged
 	}
 
-	if containerSc.RunAsUser != nil {
+	if runtime.GOOS != "windows" && effectiveSc != nil && containerSc != nil && containerSc.RunAsUser != nil {
 		effectiveSc.RunAsUser = new(int64)
 		*effectiveSc.RunAsUser = *containerSc.RunAsUser
 	}
 
-	if containerSc.RunAsGroup != nil {
+	if runtime.GOOS != "windows" && effectiveSc != nil && containerSc != nil && containerSc.RunAsGroup != nil {
 		effectiveSc.RunAsGroup = new(int64)
 		*effectiveSc.RunAsGroup = *containerSc.RunAsGroup
 	}
 
-	if containerSc.RunAsNonRoot != nil {
+	if runtime.GOOS != "windows" && effectiveSc != nil && containerSc != nil && containerSc.RunAsNonRoot != nil {
```

Comment:
I don't think this one should be overridden due to the conversation on https://github.com/kubernetes/kubernetes/pull/92355#issuecomment-654996758 where we decided to check that the `ContainerAdministrator` is not set as for a RunAsNonRoot check.